File: Pilas/Con_Listas/pila_lista.py
class Node:
    def __init__(self, data):
        self.data = data
        self.next = None

class Stack:
    def __init__(self):
        self.top = None

    def push(self, data):
        new_node = Node(data)
        new_node.next = self.top
        self.top = new_node

    def pop(self):
        if self.top is None:
            print("Pila vacía")
            return None
        data = self.top.data
        self.top = self.top.next
        return data

    def peek(self):
        return None if self.top is None else self.top.data

    def display(self):
        temp = self.top
        while temp:
            print(temp.data, "->", end=" ")
            temp = temp.next
        print("None")

# Ejemplo
p = Stack()
p.push(10)
p.push(20)
p.push(30)
p.display()
p.pop()
p.display()

File: Pilas/Con_Listas/pila_lista.cs
using System;

class Node {
    public int Data;
    public Node Next;
    public Node(int data) { Data = data; Next = null; }
}

class Stack {
    private Node top;

    public void Push(int data) {
        Node newNode = new Node(data);
        newNode.Next = top;
        top = newNode;
    }

    public int Pop() {
        if (top == null) {
            Console.WriteLine("Pila vacía");
            return -1;
        }
        int value = top.Data;
        top = top.Next;
        return value;
    }

    public int Peek() => top == null ? -1 : top.Data;

    public void Display() {
        Node temp = top;
        while (temp != null) {
            Console.Write(temp.Data + " -> ");
            temp = temp.Next;
        }
        Console.WriteLine("null");
    }

    static void Main() {
        Stack pila = new Stack();
        pila.Push(10);
        pila.Push(20);
        pila.Push(30);
        pila.Display();
        pila.Pop();
        pila.Display();
    }
}